test/support/protocol_utils.rb

```diff
@@ -230,6 +230,29 @@ def assert_locals_result expected, frame_idx: 0
       end
     end
 
+    def assert_threads_result(expected_names)
+      case ENV['RUBY_DEBUG_TEST_UI']
+      when 'vscode'
+        res = send_dap_request 'threads'
+        failure_msg = FailureMessage.new{create_protocol_message "result:\n#{JSON.pretty_generate res}."}
+
+        threads = res.dig(:body, :threads)
+
+        assert_equal expected_names.count, threads.count, failure_msg
+
+        thread_names = threads.map { |t| t[:name] }
+
+        expected_names.each do |expected|
+          thread_names.reject! do |name|
+            name.match?(expected)
```

Comment:
We can use `assert_match` method, could you tell me the reason why you implemented so?

Reply:
Why do you think we can use it here?

Reply:
Ah, that's right. My bad 🙏

Reply:
No worries 😉 
Do other changes look good to you?